### PR TITLE
Fix getSubItems in ItemLocomotive and getItemWithRenderer  to properl…

### PR DESCRIPTION
**Pull request by Forecaster**
…y display locomotive without unique renderType in creative inventory

Change how default locomotive item color is defined to make adding new locomotive easier.
Add creative locomotive that does not require any fuel, but has no recipe

### CODE REVIEW

File: src/main/java/mods/railcraft/common/modules/ModuleLocomotives.java [hunk 1]
```diff
@@ -105,6 +110,11 @@ public void initSecond() {
                     'G', RailcraftItem.gear.getRecipeObject(EnumGear.STEEL),
                     'T', RailcraftItem.plate.getRecipeObject(EnumPlate.STEEL));
         }
+
+        if (EnumCart.LOCO_CREATIVE.isEnabled()) {
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/modules/ModuleLocomotives.java]: This block doesn't do anything, remove.

File: src/main/java/mods/railcraft/common/gui/containers/ContainerLocomotiveCreative.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        for (int var1 = 0; var1 < this.crafters.size(); ++var1) {
+            ICrafting var2 = (ICrafting) this.crafters.get(var1);
+        }
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public void updateProgressBar(int id, int value) {
+        super.updateProgressBar(id, value);
+    }
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/gui/containers/ContainerLocomotiveCreative.java]: These three functions aren't doing anything, remove them.

File: src/main/java/mods/railcraft/common/carts/ItemLocomotive.java [hunk 3]
```diff
@@ -197,23 +208,25 @@ public static String getModel(ItemStack stack) {
...
-            ItemLocomotive item = (ItemLocomotive) stack.getItem();
-            if (item.renderType == LocomotiveRenderType.ELECTRIC)
-                return EnumColor.BLACK;
-            return EnumColor.GRAY;
+            return ((ItemLocomotive)stack.getItem()).getSecondaryColorF();
         }
         return EnumColor.fromId(nbt.getByte("secondaryColor"));
     }
 
+    public EnumColor getPrimaryColorF() {
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/carts/ItemLocomotive.java]: Maybe "getDefaultPrimaryColor()" is a better name?

**Comment by CovertJaguar** [comment]
Don't worry about the comments, I just went ahead and made the changes myself since they were so minor.

**Comment by Forecaster** [comment]
That's fine!

**Pull request merged**